### ISSUE TITLE
Is there a way to bypass a VPN on certain sites?

### ISSUE THREAD

**Issue by PacoH** [yt-dlp/yt-dlp#8565]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.10.13** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I started using ProtonVPN full time. The problem is, Youtube has problems with it so I have to disconnect each time I download a Youtube video. Not only is this a PITA, it is all too easy to forget to reconnect. Is there a way to just bypass the VPN on Youtube or other sites without disconnecting like this? I'm even thinking of disconnecting/reconnecting the VPN within a shell script.

### Provide verbose output that clearly demonstrates the problem

- [x] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--restrict-filenames', '-q', '--no-warnings', '--no-check-certificate', '--get-filename', '-o', '%(title)s - %(channel)s - %(id)s - %(resolution)s %(format_id)s %(vcodec)s %(acodec)s.%(ext)s', 'https://www.youtube.com/watch?v=C3_t5fXrYjw&pp=ygUbcmVjZXR0ZSBkZSBtYWdyZXQgZGUgY2FuYXJk']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8 (No ANSI), error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (pip)
[debug] Python 3.12.0 (CPython x86_64 64bit) - macOS-13.3.1-x86_64-i386-64bit (OpenSSL 3.1.4 24 Oct 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, sqlite3-3.43.2, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1890 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.10.13, Current version: stable@2023.10.13
yt-dlp is up to date (stable@2023.10.13)
[youtube] Extracting URL: https://www.youtube.com/watch?v=C3_t5fXrYjw&pp=ygUbcmVjZXR0ZSBkZSBtYWdyZXQgZGUgY2FuYXJk
[youtube] C3_t5fXrYjw: Downloading webpage
[youtube] C3_t5fXrYjw: Downloading ios player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading player 5bdfe6d5
[debug] [youtube] Unable to extract nsig function code (caused by ExtractorError('Download of https://www.youtube.com/s/player/5bdfe6d5/player_ias.vflset/en_US/base.js failed: 389386 bytes read, 386738 more expected (caused by <IncompleteRead: 389386 bytes read, 386738 more expected>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] C3_t5fXrYjw: Downloading player 5bdfe6d5
[debug] [youtube] Unable to extract nsig function code (caused by ExtractorError('Download of https://www.youtube.com/s/player/5bdfe6d5/player_ias.vflset/en_US/base.js failed: 453848 bytes read, 322276 more expected (caused by <IncompleteRead: 453848 bytes read, 322276 more expected>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] C3_t5fXrYjw: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] C3_t5fXrYjw: Downloading 1 format(s): 248+140
Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv
[debug] Command-line config: ['-vU', '--concurrent-fragments', '20', '-c', '--progress', '--no-warnings', '--no-check-certificate', '-q', '-i', '--no-mtime', '--no-playlist', '--sub-lang', 'en,en-US,en_US', '--sub-format', 'srt', '--embed-subs', '--write-auto-subs', '--restrict-filenames', '--trim-filenames', '150', '-o', 'Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv', 'https://www.youtube.com/watch?v=C3_t5fXrYjw&pp=ygUbcmVjZXR0ZSBkZSBtYWdyZXQgZGUgY2FuYXJk']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (pip)
[debug] Python 3.12.0 (CPython x86_64 64bit) - macOS-13.3.1-x86_64-i386-64bit (OpenSSL 3.1.4 24 Oct 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, sqlite3-3.43.2, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1890 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.10.13, Current version: stable@2023.10.13
yt-dlp is up to date (stable@2023.10.13)
[youtube] Extracting URL: https://www.youtube.com/watch?v=C3_t5fXrYjw&pp=ygUbcmVjZXR0ZSBkZSBtYWdyZXQgZGUgY2FuYXJk
[youtube] C3_t5fXrYjw: Downloading webpage
[youtube] C3_t5fXrYjw: Downloading ios player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading android player API JSON
[youtube] C3_t5fXrYjw: Downloading iframe API JS
[youtube] C3_t5fXrYjw: Downloading web player API JSON
[youtube] C3_t5fXrYjw: Downloading web player API JSON
[youtube] C3_t5fXrYjw: Downloading web player API JSON
[youtube] C3_t5fXrYjw: Downloading web player API JSON
[youtube] C3_t5fXrYjw: Downloading m3u8 information
[youtube] C3_t5fXrYjw: Downloading initial data API JSON
[info] C3_t5fXrYjw: Downloading subtitles: en
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] C3_t5fXrYjw: Downloading 1 format(s): 248+251
[info] Writing video subtitles to: Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv.en.vtt
[debug] Invoking http downloader on "https://www.youtube.com/api/timedtext?v=C3_t5fXrYjw&ei=eLJPZYrGENG6kwbXqoHYCg&caps=asr&opi=112496729&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1699747048&sparams=ip%2Cipbits%2Cexpire%2Cv%2Cei%2Ccaps%2Copi%2Cxoaf&signature=262E99FA3BDCD9C2D37F6B817967075855277435.E2D1BE3A7253EE1468A492120CEC911C6768033B&key=yt8&kind=asr&lang=fr&tlang=en&fmt=vtt"
[download] Destination: Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv.en.vtt
[download]    7.00KiB at    1.32MiB/s (00:00:00)[download] Got error: 0 bytes read. Retrying (1/10)...
[download] Unable to resume
[download] Destination: Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv.en.vtt
[download]  78.5% of   13.21KiB at    1.04KiB/s ETA 00:02[download] Got error: Downloaded 10622 bytes, expected 13529 bytes. Retrying (2/10)...
[download] Unable to resume
[download] Destination: Comment realiser une cuisson parfaite d un magret de canard - Oui Chef a domicile - C3 t5fXrYjw - 1920x1080 248+140 vp09.00.40.08 mp4a.40.2.mkv.en.vtt
[download]  58.1% of   13.21KiB at    786.54B/s ETA 00:07[download] Got error: Downloaded 7866 bytes, expected 13529 bytes. Retrying (3/10)...
[download] Got error: Remote end closed connection without response. Retrying (4/10)...
[download] Unable to resume
…
```

**Comment by MrRawes** [comment]
this issue should be closed and marked with external-issue, AFIAK there's nothing yt-dlp can do about this in terms of bypassing going through a vpn you could connect to a proxy but that still would be going through the vpn

**Comment by coletdjnz** [comment]
Have a look to see if the VPN app supports split tunneling, or something similar to exclude yt-dlp. 

The only mechanism yt-dlp has is for proxies with the NO_PROXY env var, but that won't work here.

This is a question to ask on the likes of the protonvpn subreddit or other help forums.

**Comment by PacoH** [comment]
I found out there is no direct support for split tunneling in macOS Ventura. The procedure to emulate it is complicated.

I did find [phx/protonvpn-cli-macos: MacOS command-line client for ProtonVPN. Written in Python](https://github.com/phx/protonvpn-cli-macos). I can connect and disconnect a particular server name from the command line. The only thing is that the app does not reflect these changes. But I usually only use `yt-dlp` from a shell script anyway so I could simply disconnect before the call and reconnect afterwards. I could write a script to display status instead of clicking on the app icon.
